### PR TITLE
refactor: 💡 Migrate host-catalogs list to `Hds::Table`

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-11726

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-11726)

## Description
- Migrated host-catalogs list to use `Hds::Table` instead of `Rose::Table`
- Migrated `Copyable` components to `Hds::Copy::Snippet` component
- Migrated `p` and unwrapped elements with `Hds::Text` component

### Before:
--- 
Screenshot 2023-12-19 at 10 42 42

### After: 
---
Screenshot 2023-12-19 at 10 42 16

## How to Test
- [ ] Click on Vercel link for Admin UI or pull down PR and run it locally using `yarn start`
- [ ] Click on any "Orgs" card created by mirage
- [ ] Click on any "Projects" card created by mirage and "Host Catalogs" will appear in the sidebar navigation.
- [ ] Click on "Host Catalogs" in the sidebar navigation and view list.
- [ ] Ensure clicking on a `host-catalog` will take you to details page view of host catalog.
- [ ] Ensure clicking on `copy` icon will copy the id of `host-catalog`

:technologist: [Admin preview](PATH_TO_FEATURE)

## Checklist:

- [x] I have added before and after screenshots for UI changes
~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
- [x] My changes generate no new warnings
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

File: ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs [hunk 1]
```diff
@@ -36,46 +36,44 @@
...
-                </Copyable>
-              </row.cell>
-            </body.row>
-          {{/each}}
-        </table.body>
-      </Rose::Table>
+                  {{B.data.displayName}}
+                </LinkTo>
+              {{else}}
+                {{B.data.displayName}}
```
Comment (calcaide) [inline ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs:58]: This string isn't wrapped in a text element

File: ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs [hunk 2]
```diff
@@ -36,46 +36,44 @@
...
+              {{/if}}
+              <p>{{B.data.description}}</p>
```
Comment (calcaide) [inline ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs:60]: Can we use [Hds::Text](https://helios.hashicorp.design/components/text) here?

File: ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs [hunk 3]
```diff
@@ -36,46 +36,44 @@
...
+            </B.Td>
+            <B.Td>
+              <HostCatalogTypeBadge @model={{B.data}} />
+            </B.Td>
+            <B.Td>
+              <Copyable
```
Comment (calcaide) [inline ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs:66]: Can we use [HDS::Copy::Button](https://helios.hashicorp.design/components/copy/button?tab=code) here?

**Review by calcaide — CHANGES_REQUESTED** [review]
Left some comments

File: ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs (hunk 1 above)
Comment (dvonanderson) [reply ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs]: @calcaide I had these changes committed yesterday failed to push them up.🙈.  Changes on next commit

File: ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs (hunk 2 above)
Comment (dvonanderson) [reply ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs]: @calcaide I had these changes committed yesterday failed to push them up.🙈.  Changes on next commit

File: ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs (hunk 3 above)
Comment (dvonanderson) [reply ui/admin/app/templates/scopes/scope/host-catalogs/index.hbs]: @calcaide I had these changes committed yesterday failed to push them up.🙈.  This one will be `Hds::Copy::Snippet`

**Review by calcaide — APPROVED** [review]
LGTM!! :rocket:

**Review by cameronperera — APPROVED** [review]

**Pull request merged**